### PR TITLE
Add server view initialization metrics

### PR DESCRIPTION
**Pull request by AmatyaAvadhanula**
Enhance logs and metrics for server view and segment metadatacache initialization

`init/serverview/time` - time in milliseconds for the coordinator's server view to be initialized
`init/serverview/time` - time in milliseconds for the broker's server view to be initialized
`init/metadatacache/time` - time in milliseconds for the broker's segment metadata cache to be initialized

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/operations/metrics.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ Metrics may have additional dimensions beyond those listed above.
 |`sqlQuery/time`|Milliseconds taken to complete a SQL query.|`id`, `nativeQueryIds`, `dataSource`, `remoteAddress`, `success`|< 1s|
 |`sqlQuery/planningTimeMs`|Milliseconds taken to plan a SQL to native query.|`id`, `nativeQueryIds`, `dataSource`, `remoteAddress`, `success`| |
 |`sqlQuery/bytes`|Number of bytes returned in the SQL query response.|`id`, `nativeQueryIds`, `dataSource`, `remoteAddress`, `success`| |
+|`broker/serverview/init/time`|Time taken to initialize the broker server view.||Depends on the number of segments|
```
Comment (abhishekagarwal87) [inline docs/operations/metrics.md:67]: maybe also add "you can use this metric to detect if your brokers are taking too long to start".

**Review by abhishekagarwal87 — APPROVED** [review]
+1. Please do look into the test failures though.

File: docs/operations/metrics.md (hunk 1 above)
Comment (AmatyaAvadhanula) [reply docs/operations/metrics.md]: Done. Thanks

**Comment by paul-rogers** [comment]
Do we have a metric for overall start time? The server view is one source of delay: there are others. If we have individual metrics, the sum won't tell us total startup delay since tasks run in parallel. Should we, in general, add a metric each time we add a lifecycle item that has to be completed before the server can be ready?

**Review by kfaraz — COMMENTED** [review]
Thanks for the changes, @AmatyaAvadhanula !

A small suggestion on the metric names:
- The service name need not be included in the name. We can always filter on node type to identify the source of the metric. So `broker/serverview/init/time` and `coordinator/serverview/init/time` can both be `init/serverView/time`.
- As @paul-rogers  suggests, we might end up adding more of these `init` times, so maybe name these something like `init/serverView/time` and `init/metadataCache/time` (or alternatively `init/time/serverView`) so that new ones would easily fit in this format.

**Review by kfaraz — APPROVED** [review]
LGTM

**Comment by AmatyaAvadhanula** [comment]
Thanks for the review @abhishekagarwal87 @paul-rogers @kfaraz 
The test failures seem unrelated
I've changed the existing metrics' names and will try to raise a separate PR with the other metrics for Lifecycle init times

**Pull request merged**